### PR TITLE
Remove unused patron email column

### PR DESCRIPTION
**Pull request by eddierubeiz**
Ref #2568 
- [x] Remove unused patron_email column from OralHistoryRequest.
- [x] Remove validation related to it
- [x] Make validation require an associated OralHistoryRequester record
- [x] in DB make the foreign key for the associated OralHistoryRequester also required

### CODE REVIEW

**Comment by eddierubeiz** [comment]
Note: I'm keeping `OralHistoryRequest#requester_email` for now, just for the sake of keeping the PR small. It's not doing any harm.

File: spec/controllers/oral_history_requests_controller_spec.rb [hunk 1]
```diff
@@ -143,40 +143,6 @@
...
-              expect(class_name).to eq "OralHistoryRequestNotificationMailer"
-              expect(action).to eq "notification_email"
-          }
-
-          expect(response).to redirect_to(work_path(work.friendlier_id))
-          expect(flash[:success]).to match /Your request will be reviewed/
-        end
-      end
-    end
-
```
Comment (jrochkind) [inline spec/controllers/oral_history_requests_controller_spec.rb:179]: So if I remember right, the old flag functionality SHOULD still work even without this column.  Removing the column ought not to remove ability to switch the feature flag back and forth at this point. So ideally these tests might remain?

Did these tests fail? Is it possibly just a test setup problem (trying to set it up with the email in the old column), and if you just set up the data in the new way the tests as written will still pass?

Comment (eddierubeiz) [reply spec/controllers/oral_history_requests_controller_spec.rb]: I am not sure why these got mistakenly removed. Glad you caught this!

**Comment by jrochkind** [comment]
The other thing we _could_ do is change the database schema to make column correspoinding to `OralHistoryRequest.oral_history_requester_id` required non-nillable.  

What do you think, is it worth it? Your call, other than that approved!

**Comment by eddierubeiz** [comment]
That migration is in here.
`change_column_null :oral_history_access_requests, :oral_history_requester_email_id, false`.

**Comment by jrochkind** [comment]
Oh nice, thank you!

**Pull request merged**